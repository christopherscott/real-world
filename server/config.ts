const {
  NODE_ENV,
  PRISMA_ENDPOINT,
  PRISMA_TEST_ENDPOINT,
  PRISMA_SECRET,
  APP_SECRET,
  TMDB_KEY,
  OMDB_KEY,
} = process.env;

const isTest = NODE_ENV === 'test';

export default {
  PRISMA_ENDPOINT: isTest ? 'http://prisma.com' : PRISMA_ENDPOINT,
  PRISMA_TEST_ENDPOINT: isTest ? 'http://prisma.com' : PRISMA_TEST_ENDPOINT,
  PRISMA_SECRET: isTest ? '123' : PRISMA_SECRET,
  APP_SECRET: isTest ? '123' : APP_SECRET,
  TMDB_KEY: isTest ? '123' : TMDB_KEY,
  OMDB_KEY: isTest ? '123' : OMDB_KEY,
};
