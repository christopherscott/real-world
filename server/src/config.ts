export const PRISMA_ENDPOINT =
  process.env.NODE_ENV === 'test'
    ? 'http://localhost:4467'
    : process.env.PRISMA_ENDPOINT;

export const PRISMA_SECRET = process.env.PRISMA_SECRET;
