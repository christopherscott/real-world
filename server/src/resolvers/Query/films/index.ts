import { Context } from '../../../utils';
import { Film } from '../../../generated/types';

export const films = (parent, args, context: Context, info) => {
  // Order by multiple fields is not supported yet
  // https://github.com/prismagraphql/prisma/issues/62
  return context.db.query.films({ orderBy: 'rating_DESC' }, info);
};

export const film = (
  parent,
  { id }: { id: Film['id'] },
  context: Context,
  info
) => {
  return context.db.query.film({ where: { id } }, info);
};
