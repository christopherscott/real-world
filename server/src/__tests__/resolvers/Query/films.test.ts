import { map } from 'ramda';
import { db, client } from '../../config';
import films from './fixtures/films';

beforeEach(async () => {
  await Promise.all(map(data => db.mutation.createFilm({ data }), films));
});

afterEach(async () => {
  await db.mutation.deleteManyFilms({});
});

test('should return films ordered by raiting', async () => {
  const query = `
    query {
      films {
        title
        rating
      }
    }
  `;

  const responseData = await client.request(query);

  expect(responseData).toMatchSnapshot();
});
