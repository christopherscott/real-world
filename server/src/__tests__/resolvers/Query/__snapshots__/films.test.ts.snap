// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should return films ordered by raiting 1`] = `
Object {
  "films": Array [
    Object {
      "rating": 9.3,
      "title": "The Shawshank Redemption",
    },
    Object {
      "rating": 9.2,
      "title": "The Godfather",
    },
    Object {
      "rating": 9,
      "title": "The Godfather: Part II",
    },
    Object {
      "rating": 9,
      "title": "Pulp Fiction",
    },
    Object {
      "rating": 9,
      "title": "The Good, the Bad and the Ugly",
    },
  ],
}
`;
